tests/lib/c_lib/src/main.c

```diff
@@ -618,6 +618,15 @@ void test_asctime(void)
 	zassert_true(strcmp(asctime(&time), "??? ???  3 02:01:00 1905\n") == 0, "asctime error");
 }
 
+void test_div(void)
+{
+	/* test example from man */
+	div_t q = div(-5, 3);
```

Comment:
Please extend this to cover all combinations of positive and negative arguments, and for each one verify that the result invariant is correct.  Something like:
```
struct input_s {
   int num;
   int den;
} const input[] = {
   { 5, 3 },
   { 5, 7 },
   { -5, 7 },
   { 5, -7 },
   { -5, -7 },
};

for (size_t i = 0; i < ARRAY_SIZE(input); ++i) {
   int num = input[i].num;
   int den = input[i].den;
   
   div_t res = div(num, den);
   zassert_equal(num, res.quot * den + rem,
                 "Invariant failed for %d / %d : q = %d, r = %d",
		 num, den, res.quot, res.rem);
   zassert_equal(num >= 0, res.rem >= 0,
                 "Quotient not rounded toward zero : q = %d, r = %d",
		 num, den, res.quot, res.rem);
}
```
Also provide a similar test for `ldiv`.

Reply:
about the commit messages, I'll do it.
about adding more tests, i saw this at the top of the file:

` IMPORTANT: The module only ensures that each supported library is present,
 * and that a bare minimum of its functionality is operating correctly. It does
 * NOT guarantee that ALL standards-defined functionality is present, nor does
 * it guarantee that ALL functionality provided is working correctly.`

my guess was that they try to minimize tests for simplicity of the file and they allowed it beacuse the functions logics which this file tests is pretty simple. 
of course I would add it if you still want me to.

Reply:
I don't think that's justifying minimizing tests for provided functions; it's explaining that not all functions are supported.  Please do provide full testing for any function added.